sm2/signature.go

```diff
@@ -183,7 +191,7 @@ func Sign(rand io.Reader, priv *ecdsa.PrivateKey, id string, msg []byte, hasher
 		if opt, ok := priv.Curve.(invertible); ok {
 			d1Inv = opt.Inverse(d1)
 		} else {
-			d1Inv = new(big.Int).ModInverse(d1, N)
+			d1Inv = fermatInverse(d1, N)
```

Comment:
This function will not be invoked under the amd64 platform for the `sm2p256v1` curve.